content/automate/ManageIQ/Service/Retirement/StateMachines/Methods.class/__methods__/check_service_retired.rb

```diff
@@ -1,37 +1,72 @@
-#
-# Description: This method checks to see that all of the service resources are retired before retiring the service.
-#
-
-service = $evm.root['service']
-if service.nil?
-  $evm.log('error', "Service Object not found")
-  exit MIQ_ABORT
-end
+module ManageIQ
+  module Automate
+    module Service
+      module Retirement
+        module StateMachines
+          module Methods
+            class CheckServiceRetired
+              def initialize(handle = $evm)
+                @handle = handle
+              end
+
+              def main
+                service = $evm.root['service']
+                if service.nil?
+                  @handle.log('error', "Service Object not found")
+                  exit MIQ_ABORT
+                end
+
+                @handle.log('info', "Checking if all service resources have been retired.")
+                result = 'ok'
+
+                service.service_resources.each do |sr|
+                  check_service_resource_retirement(sr)
+                end
 
-$evm.log('info', "Checking if all service resources have been retired.")
+                @handle.log('info', "Service: #{service.name} Resource retirement check returned <#{result}>")
+                case result
```

Comment:
@d-m-u You could put the result logic in a method instead of doing it inline here.